Document/0x05b-Basic-Security_Testing.md

```diff
@@ -534,7 +534,99 @@ Rooting an emulator is therefore unnecessary; root access can be established wit
+
+```
+$ frida-ps -U | grep -i owasp
+21228  sg.vp.owasp_mobile.omtg_android
+
+$ objection -g sg.vp.owasp_mobile.omtg_android explore
+
+...[usb] # cd ..
+/data/user/0/sg.vp.owasp_mobile.omtg_android
+
+...[usb] # ls
+Type       ...  Name
+---------  ...  -------------------
+Directory  ...  cache
+Directory  ...  code_cache
+Directory  ...  lib
+Directory  ...  shared_prefs
+Directory  ...  files
+Directory  ...  app_ACRA-approved
+Directory  ...  app_ACRA-unapproved
+Directory  ...  databases
+
+Readable: True  Writable: True
+
+```
+
+One you have a file you want to download you can just run `file download some_file`. This will download that file to your working directory. The same way you can upload files using `file upload`.
+
+```
+...[usb] # ls
+Type    ...  Name
+------  ...  -----------------------------------------------
+File    ...  sg.vp.owasp_mobile.omtg_android_preferences.xml
+
+Readable: True  Writable: True
+...[usb] # file download sg.vp.owasp_mobile.omtg_android_preferences.xml
+Downloading ...
+Streaming file from device...
+Writing bytes to destination...
+Successfully downloaded ... to sg.vp.owasp_mobile.omtg_android_preferences.xml
+
+```
+
+The downside is that, at the time of this writing, objection does not support bulk file transfer yet, so you're restricted to copy individual files. Still, this can come handy in some scenarios where you're already exploring the app using objection anyway and find some interesting file. Instead of e.g. taking note of the full path of that file and use `adb pull path_to_some_file` from a separate terminal, you might just want to directly do `file download some_file`.
+
+##### Using Termux
+
+If you have a rooted device and have [Termux](https://play.google.com/store/apps/details?id=com.termux "Termux on Google Play") installed, you can start a FTP server.
+
+Install and run `tsu` on the Android device, which is the Termux su binary: `pkg install tsu`
+
+Now you can run `tcpsvd` as root:
+
+```
+$ tsu
+# tcpsvd -vE 0.0.0.0 1024 ftpd /
+```
+
+![FTP from Android](Images/Chapters/0x05b/android-ftp-from-android.png)
+
+This will open FTP access to the root folder of the device (`/`) on port 1024. You can now use any command line FTP client or, for example, if you're on macOS you can type `cmd+k` from Finder and give in the IP of the Android device, e.g. `ftp://192.168.1.108:1024`
+
+![FTP from macOS](Images/Chapters/0x05b/android-ftp-from-macos.png)
+
+Remember that FTP is not encrypting its communication and is therefore considered insecure. For projects classified as high risk or in critical environments you may want to transfer data preferably via the USB cable and not unencrypted via the network. Check the [Termux Wiki](https://wiki.termux.com/wiki/Remote_Access "Termux Remote Access") to learn more about remote file access methods.
```

Comment:
Seems a bit overkill.

What would make more sense is "remember that you are publicly exposing your entire device to anyone on the same network as you". The fact that FTP is an insecure channel seems like a wrong threat model to tackle. @commjoen  / @sushi2k ?

Reply:
Not overkill to me, rather: make them use FTP over SSH instead of teh usb stuff, because that is more work..especially if your device farm is somewhere else...

Reply:
Carlos will try to use SFTP instead of FTP and update the descrption.

Reply:
That solves both aspects 👍